app/javascript/lib/validation/shared-validation.js

```diff
@@ -0,0 +1,184 @@
+      const labelColumnIndex = headers[0].indexOf(`${colName }__ontology_label`)
+      if (excludedColumns.includes(colName) ||
+          colName.endsWith('ontology_label') ||
+          headers[1][index] === 'numeric' ||
+          labelColumnIndex === -1) {
+        return null
+      }
+      // for each column, track a hash of label=>value,
+      // and also a set of mismatched values--where the same label is used for different ids
+      return { colName, index, labelColumnIndex, labelValueMap: {}, mismatchedVals: new Set() }
+    }).filter(c => c)
+  }
+  // for each column we need to check, see if there is a corresponding filled-in label,
+  //  and track whether other ids have been assigned to that label too
+  for (let i = 0; i < dataObj.dragCheckColumns.length; i++) {
+    const dcc = dataObj.dragCheckColumns[i]
+    const colValue = line[dcc.index]
+    const label = line[dcc.labelColumnIndex]
+    if (label.length) {
+      if (dcc.labelValueMap[label] && dcc.labelValueMap[label] !== colValue) {
+        dcc.mismatchedVals.add(label)
+      } else {
+        dcc.labelValueMap[label] = colValue
+      }
+    }
+  }
+  // only report out errors if this is the last line of the file so that a single, consolidated message can be displayed per column
+  if (isLastLine) {
+    dataObj.dragCheckColumns.forEach(dcc => {
+      if (dcc.mismatchedVals.size > 0) {
+        const labelString = [...dcc.mismatchedVals].slice(0, 10).join(', ')
+        const moreLabelsString = dcc.mismatchedVals.size > 10 ? ` and ${dcc.mismatchedVals.size - 10} others`: ''
+        issues.push(['error', 'content:metadata:mismatched-id-label',
+          `${dcc.colName} has different ID values mapped to the same label.
+          Label(s) with more than one corresponding ID: ${labelString}${moreLabelsString}`])
+      }
+    })
+  }
+  return issues
+}
+
+
+/** raises a warning if a group column has more than 200 unique values */
+export function validateGroupColumnCounts(headers, line, isLastLine, dataObj) {
+  const issues = []
+  const excludedColumns = ['NAME']
+  if (!dataObj.groupCheckColumns) {
+    dataObj.groupCheckColumns = headers[0].map((colName, index) => {
+      if (excludedColumns.includes(colName) || colName.endsWith('ontology_label') || headers[1][index] === 'numeric') {
+        return null
+      }
+      return { colName, index, uniqueVals: new Set() }
+    }).filter(c => c)
+  }
+  for (let i = 0; i < dataObj.groupCheckColumns.length; i++) {
+    const gcc = dataObj.groupCheckColumns[i]
+    const colValue = line[gcc.index]
+    if (colValue) { // don't bother adding empty values
+      gcc.uniqueVals.add(colValue)
+    }
+  }
+
+  if (isLastLine) {
+    dataObj.groupCheckColumns.forEach(gcc => {
+      if (gcc.uniqueVals.size > 200) {
+        issues.push(['warn', 'content:group-col-over-200',
+            `${gcc.colName} has over 200 unique values and so will not be visible in plots -- is this intended?`])
```

Comment:
Fixes indentation to be 2-space, etc.:

```suggestion
        issues.push([
          'warn', 'content:group-col-over-200',
          `${gcc.colName} has over 200 unique values and so will not be visible in plots -- is this intended?`
        ])
```